seisbench/models/base.py

```diff
@@ -339,6 +447,7 @@ def __init__(
         self._annotate_functions = self._annotate_function_mapping.get(
             output_type, None
         )
+        self._argspec = locals()
```

Comment:
I would be slightly apprehensive about making the users overwrite an abstract method to save any new model. I think it it just another step which adds complexity to our model API. I may have a fix for this to do it automatically, it is through object introspection, and slightly verbose but it does manage to avoid the issue of users having to perform extra steps in order to save a model.